### PR TITLE
Add Robots and Marbles tutorial part 7 - Parameter Sweeping

### LINKED ISSUE #79: Parameter Sweeping introduced in the Robot and Marbles tutorial series
## Description
We want to continue the Robot and Marbles tutorial series with a simple tutorial of the Parameter Sweeping feature found in cadCAD. 

Parameter Sweeping is a cadCAD feature that runs a model under different possible parameterizations. In a cadCAD model, parameters might refer to intrinsic settings of the system being designed that are still subject to change. They might also refer to assumptions made about properties of the agents that interact with the system, for example. Parameter sweeping is especially helpful in sensitivity analysis, where we are interested in exploring the effects of changing a parameter, or if the desired goals of the system are achieved under a range of assumptions about external agents. 

## Instructions
1. Fork the Robot and Marbles part 4 notebook and name it Robot and Marbles Part 7 - Parameter Sweeps.
2. Modify it to make use of the Parameter Sweep feature where the `robots_probabilities` global constant is turned into a parameter to be swept. (Similar to part 6 being a modified fork of part 4 introducing the A/B Testing feature),
3.  Compare the results of the sweeps with plots (as was done in Robot and Marbles part 6)

## Acceptance Criteria

- `robots_probabilities` is parameterized
- Variable and function naming conventions match those in the previous robot and marbles tutorials
- The difference in the resulting datasets is clearly visible in the plots
- Plot Style is consistent with earlier robot and marbles tutorials
- The range of the parameter sweep should be appropriate for obvious difference in the resulting datasets
- Documentation assumes readers have completed Parts 1 through 6, and builds on these materials:
- Markdown cells throughout the notebook guiding  the reader through the feature
- Code cells contain comments highlighting relevant pieces of the code

## References

- https://github.com/BlockScience/cadCAD/blob/master/tutorials/robot-marbles-part-4/robot-marbles-part-4.ipynb
- https://github.com/BlockScience/cadCAD/blob/master/tutorials/robot-marbles-part-6/robot-marbles-part-6.ipynb
- https://github.com/BlockScience/cadCAD/blob/master/documentation/examples/param_sweep.py

### PR DESCRIPTION
**Pull request by sembrestels**
This pull request closes #79. You can review the [notebook here](https://github.com/BlockScience/cadCAD/blob/d0d29a589da2b0e0c83b6d580833458a252770b6/tutorials/robot-marbles-part-7/robot-marbles-part-7.ipynb).

To be determined:

- [x] Is `capacity` a good param for this case? Do we add more params or is it better to keep it simple?
- [x] Parameters are a list or an object depending on how much configurations. [Is it ok?](https://community.cadcad.org/t/why-are-parameters-are-a-list-on-single-proc-mode/128)
- [ ] Review texts

### CODE REVIEW

**Comment by mzargham** [comment]
The text overall looks pretty good but there some recommended changes, mostly in the form of making what you did clearer for the reader:
1. capacity is a reasonable parameter to sweep but you need to write out the definition, e.g. 'in the preceding example each robot arm could move exactly 1 marble, in this tutorial we call it's *capacity*
2. in accordance with the above, i would add the variable in the top version of the simulation and simply set it to 1
3. It is important that people understand that there can be multiple parameters, and i think *capacity* is logically a per robot variable, as such I would recommend making the capacity parameter a dictionary as follows:  
`params = {capacity : [{'A':1, 'B':1 },{'A':1, 'B':2 }, {'A':2, 'B':2 } ]}`
and to include it in the config as `'M':params`
this change would necessitate the reference it in each robots policy as
`capacity = params['capacity']['A']  ` and `capacity = params['capacity']['B']` respectively.
4. i think there should be an extra block of markdown remarking on how the parameters impacts the results. what we will see is different limit cycles, and in particular the capacity=2 case converges to its limit cycle faster but has 2-8 variation, whereas the capacity =1 case converges slower and as tighter 4-6 variation.

**Comment by sembrestels** [comment]
I love the recommendations, specially the multiple parameters, thanks!

File: tutorials/README.md [hunk 1]
```diff
@@ -7,5 +7,6 @@ In this series, we introduce basic concepts of cadCAD and system modelling in ge
 [Part 4](robot-marbles-part-4/robot-marbles-part-4.ipynb) - Uncertainty and Stochastic Processes  
 [Part 5](robot-marbles-part-5/robot-marbles-part-5.ipynb) - Using class objects as state variables  
 [Part 6](robot-marbles-part-6/robot-marbles-part-6.ipynb) - A/B testing  
+[Part 7](robot-marbles-part-6/robot-marbles-part-7.ipynb) - Parameter Sweeping
```
Comment (markusbkoch) [inline tutorials/README.md]: typo in folder name

File: tutorials/robot-marbles-part-7/robot-marbles-part-7.ipynb [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
...ghP5tXXTvugnZuz34NCDS6iKVHgx3xtVVVW+RIQx64Annuf3UUgnH2uH2jKqmX6KnSKFrDojFaSoq6jx0eFwRn8sQLa7NJ91R35UIKpZvAL352xP9vwH71/Qb810VEQqnJniTvF8hRR+5tknNxm/dTEd/mmE20BLRCIRiw5vYIZv4oYfbXVNQpYeQ1MzeS2YLium8OJzIG2rNsB1gIXCMcRL4jFcM4MnH99wbJIrVYuVRSwWU2ZGzyR983O6j2Wu/liF759ejj68zKRdLujXboFjX9ni8PFHV+m9ahqR4mgAutRoit8Ep2xdqJRIJHRxm8KJx7K1Dl1NbfYZL6HdP42yLQaAtyEfsfRxZdeNk5QrWBwfo8X0+t+/f10JJjlyk3sGEcfH0W+1La9Dgjg5eQUVi5RS+B7eN0+x6MhGAPwtPSmYNx+rzvkww3cVYdGRmLbpxzz9cal+adScN4hvsbJOkdCl6ZeqTY0y03vwLvST/LWqSIV494spnh8aFU5Th7HESyVsHDGbJhXT3jH5M99nQJWKlGL3uEV8DPlM+bl9kEgT6/s7xyzEsGHKm3bMdy/j+OOrlC9UghOTFTeVEASBSbucOfboCuuGzaBttQb0X2PLnoRaMkCd0lW4M2OzwvdWhAfvX2C4ZgY1S1Rg2+j5hEVF0MJxPI8/vgagVP4inLfwpELhklkax884HNuM7T4P+aKtVYehLE7FKCYriI0T43RiG3ZHNiAVBOZ2H4t1p6HkyWYRvJxMbnLPAIIgMGG7A6ef3GDzyDlK1XQffXhF/zUzAHAfaIVEkNJo8ShuvX1Cm6r1cTU0p3ap1OunQ7xm8zDoJQBXp3mhrZ36jsm0mL1/FQsSFmS/c2bKSv5NpZVOKpVS1244EbFRWHYYzPCm3RR+bkhkGPorLREhwm+CE/XshvMq5IP8/e61muM30TnVe2y45IfLqR3oampzy1a5XbArTu/C45wP1p2GUlBbFxWTZggJc5c86hq8mOtN8YJZu5vxU3gI+u4WaGto4TfBicm7XFh/SbZgraGqzpohNgxv1j1LY/iZG68f0sl1CiFRMjmJphVqcmqKW4r7L7IKv3v+TNm9jOfB7+hd91+c+02mfKHs/YL7E8hN7hnA8fhW1l08wMyuoxjapKvC10dER1DXTiYGNrhhJy6+uMeEHUspXaAoO8YswLBBhzSnlx7n
```
Comment (markusbkoch) [inline tutorials/robot-marbles-part-7/robot-marbles-part-7.ipynb]: In line with my comment in the community forum, I would recommend directing readers to using the general structure for parameter sweeping and `multi_proc` from the beginning, even though there's no sweeping at this point (ie, the parameter list would contain a single item).

Comment (markusbkoch) [reply tutorials/robot-marbles-part-7/robot-marbles-part-7.ipynb]: For some reason GitHub's main PR UI is not displaying this comment in the right context (showing lines 299-302 for me, but comment refers to line [409](https://github.com/BlockScience/cadCAD/pull/88/files/d0d29a589da2b0e0c83b6d580833458a252770b6#r384627161)

**Comment by sembrestels** [comment]
You can find the new notebook [here](https://github.com/BlockScience/cadCAD/blob/b953084942847ef7d2f89333496089656cf187ac/tutorials/robot-marbles-part-7/robot-marbles-part-7.ipynb).

Could you review if setting up an extra parameter for `robot_arm(params, step, sL, s, capacity=1)` is a good practice in cadCAD simulations? I think it's pretty straight forward, but not sure if it should be done in another way (inside `params` maybe?)

**Comment by markusbkoch** [comment]
lgtm. Merging into a feature branch. Thanks, @sembrestels ! 👏

**Comment by sembrestels** [comment]
Yeah thanks for merging! And sorry for being a bit late on the delivery, it costed me a bit more than expected, but we finally have it!!

**Pull request merged**